tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/Search.cs

```diff
@@ -284,136 +285,156 @@ public SearchResponseVM GetPinnedData(SearchRequestVM searchRequestVM, string li
+                                        }
+                                        else
+                                        {
+                                            searchResponse.DocumentDataList = userpinnedDocumentCollection.Values;
+                                        }
                                     }
-                                    else
+                                    IList<DocumentData> filterDocumentList = null;
+                                    if (!string.IsNullOrWhiteSpace(searchRequestVM.SearchObject.Filters.AreaOfLaw))
                                     {
-                                        searchResponse.DocumentDataList = userpinnedDocumentCollection.Values;
+                                        filterDocumentList = userpinnedDocumentCollection.Values.Where(pinDocument => searchRequestVM.SearchObject.Filters.AreaOfLaw.Contains(pinDocument.DocumentClient)).ToList();
+                                    }
+                                    if (filterDocumentList != null)
+                                    {
+                                        searchResponse.DocumentDataList = filterDocumentList;
+                                        searchResponse.TotalRows = filterDocumentList.Count();
                                     }
-                                }
-                                IList<DocumentData> filterDocumentList = null;
-                                if (!string.IsNullOrWhiteSpace(searchRequestVM.SearchObject.Filters.AreaOfLaw))
-                                {
-                                    filterDocumentList = userpinnedDocumentCollection.Values.Where(pinDocument => searchRequestVM.SearchObject.Filters.AreaOfLaw.Contains(pinDocument.DocumentClient)).ToList();
-                                }
-                                if (filterDocumentList != null)
-                                {
-                                    searchResponse.DocumentDataList = filterDocumentList;
-                                    searchResponse.TotalRows = filterDocumentList.Count();
                                 }
                             }
                             else
                             {
                                 Dictionary<string, MatterData> userpinnedMatterCollection =
                                     JsonConvert.DeserializeObject<Dictionary<string, MatterData>>(userPinnedMatter);
                                 searchResponse.TotalRows = userpinnedMatterCollection.Count;
-                                // searchResponse.MatterDataList = userpinnedMatterCollection.Values.Reverse();
-                                if (searchRequestVM.SearchObject.Sort.Direction == 0)
+                                if (searchRequestVM.SearchObject.Sort.SortAndFilterPinnedData == false)
```

Comment:
If SortAndFilterPinnedData == false just return the pinned results with out sorting and filtering else sort or filter the pinned data